vernac/metasyntax.ml

```diff
@@ -1654,10 +1654,16 @@ let add_syntactic_definition ~local deprecation env ident (vars,c) { onlyparsing
 (**********************************************************************)
 (* Declaration of custom entry                                        *)
 
+let warn_custom_entry =
+  CWarnings.create ~name:"custom-entry-overriden" ~category:"parsing"
+         (fun s ->
+          strbrk "Custom entry " ++ str s ++ strbrk " has been overriden.")
+
 let load_custom_entry _ _ = ()
 
 let open_custom_entry _ (_,(local,s)) =
-  Egramcoq.create_custom_entry ~local s
+  if Egramcoq.exists_custom_entry s then warn_custom_entry s
```

Comment:
Isn't the warning bogus? For example, if you do `Import M. Import M.` where `M` contains a custom entry declaration, it will be triggered, but should not.

Reply:
Yes, but I think we have the same problem with notations, up to a sordid heuristic that tries to recognize that two notations are the same. The proper fix is to use kernames, which I am implementing.

Reply:
(Also, try pondering for one minute what happens if you have custom entries in functors / diamond inheritance via `Export` to realize that there is no way to get a sound and complete warning in the current implementation.)

Reply:
> (Also, try pondering for one minute what happens if you have custom entries in functors / diamond inheritance via `Export` to realize that there is no way to get a sound and complete warning in the current implementation.)

Oh sure, I'm aware of it, we had the same problem for notations indeed.